File: Gateway 1.2/ride.py
fileName="""
LANG: PYTHON3
TASK: ride
""".split()[-1]


def enc(s):
    code = 1
    for c in s:
        code *= ord(c) - 64
    return code % 47


r1, r2 = open('%s.in' % fileName).readlines()

with open('%s.out' % fileName, 'w') as fout:
    fout.write('GO\n' if enc(r1) == enc(r2) else 'STAY\n')
    
    ## return  a ? b : c;
    ## return  b if a else c 

    
# fout = open('%s.out' % fileName, 'w')
# fout.write(...)
# fout.close()

File: Gateway 1.2/ride.java
/*
LANG: JAVA
TASK: ride
*/

import java.io.File;
import java.io.FileWriter;
import java.util.Scanner;

public class ride {
    public static void main(String[] args) throws Exception {

        Scanner sc = new Scanner(new File("ride.in"));
        String cometName = sc.nextLine();
        String groupName = sc.nextLine();

        int code = 1;
        for (char c : cometName.toCharArray()) {
            int thisLetterCode = c - 'A' + 1;
            code = code * thisLetterCode;
        }
        int result = code % 47;

        int code2 = 1;
        for (char c : groupName.toCharArray()) {
            int thisLetterCode = c - 64;
            code2 = code2 * thisLetterCode;
        }
        int result2 = code2 % 47;

        FileWriter fw = new FileWriter("ride.out");
        // PrintWriter pw = new PrintWriter(new File("ride.out"));
        // pw.print()
        // pw.println()
        if (result == result2) {
            System.out.println("GO");
            fw.write("GO\n");
        } else {
            System.out.println("STAY");
            fw.write("STAY\n");
        }
        fw.close();
    }
}
